WordPress/create-block-theme
# Refactor GitHub release workflows

Refactor GitHub release workflows in order to comply with branch protection rules.
Also removes google fonts related workflow.

#### The new process

1. Run the "Create new release PR" manually, selecting the release type ("major", "minor", or "patch"). This will create a new branch including the Changelog update and the version bumo on the readme, create-block-theme.php, and package.json files.
2. Merging the PR created on the previous step will trigger the deploy workflow, which remains unchanged, except for the conditions added in order for it to run only if the merged PR matches the characteristics of an automated release PR.



#### Testing instructions
1. Fork this repo
2. Checkout this branch
4. Merge intro trunk on the fork
5. Run the workflow, a new PR should be created on the fork
6. Merge the PR; the automated deploy step will run. At this point, the only things that you'll be able to test on the fork is whether the deploy workflow correctly identifies the PR and branch as a new release, and the tag creation. The deploy itself requires credentials only available on the parent repo.
